tensorboard/webapp/widgets/line_chart_v2/line_chart_component.scss

```diff
@@ -28,6 +30,13 @@ limitations under the License.
     'dot xaxis';
   grid-template-columns: 50px 1fr;
   grid-auto-rows: 1fr 30px;
+
+  // Unlike other Angular components, we may use this component outside
+  // of the Angular app and want to control the dark mode styling based on
+  // the component prop alone.
```

Comment:
I wasn't aware of any restriction that the LineChartComponent Angular component must be reusable outside of the Angular app.  I was expecting we could use one of the existing CSS patterns `@tb-theme-foreground-prop(color, text)` or the `@include tb-dark-theme` instead of having a 3rd way.

iiuc, TB surfaces that do not support dark theme will never add ".dark-mode" to the body, so ":host-context(.dark-mode)" would be false for them, and this line chart would behave as expected.

- Why wouldn't `@include tb-dark-theme` work outside of the Angular app?
- Are we setting a precedent that all components under widgets/ must not depend on `tensorboard/webapp/theme/tb_theme`?  (which is not the case today)

Reply:
> Why wouldn't @include tb-dark-theme work outside of the Angular app?

In a light themed app, I want to be able to provide a dark chart (since it is a prop anyways), so in that case, the `:host-context()` method is definitely not the way to go.

> Are we setting a precedent that all components under widgets/ must not depend on tensorboard/webapp/theme/tb_theme? (which is not the case today)

No, I am not setting such precedence. I would be using themed variable or mat-* variables for something that does not need to vary based on theme.